### PR TITLE
Deprecate NelmioApiDoc 2 support

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

NelmioApiDocBundle 3 has native support for API Platform. Deprecating our support for v2.

### CODE REVIEW

**Comment by soyuka** [comment]
fix #1299 ?

Adding those annotations didn't work, I think we should use `trigger_error` in the constructor instead of in the file.

File: tests/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProviderTest.php [hunk 1]
```diff
@@ -32,6 +32,8 @@
 
 /**
  * @author Teoh Han Hui <teohhanhui@gmail.com>
+ *
+ * @group legacy
```
Comment (meyerbaptiste) [inline tests/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProviderTest.php]: We should also add a `@expectedDeprecation`.

Comment (soyuka) [reply tests/Bridge/NelmioApiDoc/Extractor/AnnotationsProvider/ApiPlatformProviderTest.php]: done

**Comment by dunglas** [comment]
Thanks @soyuka

**Pull request merged**